teps/0101-env-in-pod-template.md

```diff
@@ -0,0 +1,407 @@
+of environment variables, which are combined with those defined in steps and `stepTemplate`, and then passed to all step containers.
+That allows to exclude common variables to the global level as well as overwrite defaults specified
+on the particular step level.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this TEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a TEP within the wider Tekton community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+One of the most important motivators for this feature is the ability to eliminate redundant code from
+the `PipelineRun` as well as `TaskRun` specification.
+In case of complex pipelines, which consist of dozen or even hundreds of tasks, any kind of repetition
+significantly impacts the final size of `PipelineRun` and leads to resource exhaustion.
+
+Besides, users quite frequently are willing to overwrite environment variable values
+specified in a `stepTemplate` in the single place when running pipelines.
+That helps to optionally pass settings, without a need to define additional pipeline parameters.
+
+Having an `env` field in the pod template allows to:
+
+- specify global level defaults, what is important to reduce the size of `TaskRun` and `PipelineRun`
+- override defaults from `stepTemplate` at `TaskRun` and `PipelineRun` level
+
+### Goals
+
+<!--
+List the specific goals of the TEP.  What is it trying to achieve?  How will we
+know that this has succeeded?
+-->
+
+1. The main goal of this proposal is to enable support for specification of environment variables on the global level
+(`TaskRun` and `PipelineRun`).
+
+2. Environment variables defined in the Pod template at `TaskRun` and `PipelineRun` level
+   take precedence over ones defined in steps and `stepTemplate`.
+
+### Non-Goals
+
+<!--
+What is out of scope for this TEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+### Use Cases
+
+<!--
+Describe the concrete improvement specific groups of users will see if the
+Motivations in this doc result in a fix or feature.
+
+Consider both the user's role (are they a Task author? Catalog Task user?
+Cluster Admin? etc...) and experience (what workflows or actions are enhanced
+if this problem is solved?).
+-->
+
+1. In the first case, common environment variables can be defined in a single place on a `PipelineRun` level.
+   Values can be specified as literals or through references.
+   Variables defined on a `PipelineRun` or `TaskRun` level are then available in all steps.
+   That allows to significantly reduce the size of `PipelineRun` and `TaskRun` resource,
+   by excluding the common environment variables like: static global settings, common values coming from metadata, ...
```

Comment:
Are you running into size limits for both `PipelineRuns` and `TaskRuns`? 

For `PipelineRuns`, [TEP-0100](https://github.com/tektoncd/community/blob/main/teps/0100-embedded-taskruns-and-runs-status-in-pipelineruns.md) that we're implementing should help address this problem -- wondering if you'll still face this problem even with the minimal status?

Reply:
TEP-0100 definitely can help reduce the `PipelineRuns` size, but the benefit we can get from this TEP is also the flexibility to update any global container configurations on the Run level. So that users doesn't have to go back to each `stepTemplate` to modify for every small changes.